test/doc/websocket_2_handshaking.cpp

```diff
@@ -132,6 +132,94 @@ snippets()
 
     //]
     }
+
+    {
+        //[code_websocket_2_6
+
+        // a function to select the most preferred protocol from a comma-separated list
+        auto best_protocol = [](string_view offered_tokens) -> std::string
+        {
+            // tokenize the Sec-Websocket-Protocol header offered by the client
+            http::token_list offered( offered_tokens );
+
+            // an array of protocols supported by this server
+            // in descending order of preference
+            static const std::array<string_view, 3>
+                supported = {
+                "v3.my.chat",
+                "v2.my.chat",
+                "v1.my.chat"
+            };
+
+            std::string result;
```

Comment:
Why `std::string` instead of `string_view`?

Reply:
Is this because it has to be bound into the decorator? You really need to add comments explaining this. If there was ever a need for comment, this is an obvious case of it.

Reply:
because the passing copies of string_views around seems like a really bad idea